crypto/include/datatypes.h

```diff
@@ -278,7 +278,7 @@ void octet_string_set_to_zero(void *s, size_t len);
 #define be64_to_cpu(x) bswap_64((x))
 #else /* WORDS_BIGENDIAN */
 
-#if defined(__GNUC__) && defined(HAVE_X86)
+#if defined(__GNUC__) && defined(HAVE_X86) || defined(__x86_64__)
```

Comment:
Does && go with GNUC or with the || __x86_64__?  I assume this is intended to be `defined(__GNUC__) && (defined(HAVE_X86) || defined(__x86_64__))?

Reply:
yip, I can add the parentheses to clarify